ceph/rados/rados_scrub.py

```diff
@@ -189,3 +190,34 @@ def set_osd_flags(self, flag, value):
             return True
         log.error(f"Failed to {flag} OSD flag {value}")
         return False
+
+    def get_dump_scrubs(self, osd_id):
+        """
+        Method is used to return the osd dump scrubs
+         Args:
+             osd_id: osd id number
+        Returns:
+              Dump scrub output in the json format.
+        """
+        base_cmd = f"cephadm shell --name osd.{osd_id} ceph daemon  osd.{osd_id}  dump_scrubs -f json 2>/dev/null"
+        acting_osd_node = self.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        out_put_tuple = acting_osd_node.exec_command(sudo=True, cmd=base_cmd)
+        out_put = tuple(x for x in out_put_tuple if x)
+        return json.loads(out_put[0])
+
+    def get_pg_dump_scrub(self, osd_id, pg_id):
+        """
+        Method is used to retrive the dump scrub of a pg
+        Args:
+             osd_id: osd id number
+             pg_id: pg id
+        Returns:
+              Dump scrub output of a PG in the json format.
+
```

Comment:
As per the method doc text, the intention is to retrieve the dump scrub for a particular PG, imho, it should accept only 1 input which is the PG ID, then we retrieve the acting set for that PG -> use the primary OSD ID and get its dump scrub data
This way we could eliminate the additional block of code to check whether the pg_id is part of dump scrub or not.
Please correct me if my understanding of the need of this method is wrong, in which case we should update the doc text accordingly

Reply:
It is not just the primary OSD from which I want to obtain the PG dump scrub. To obtain the PG dump scrub on any OSD on any node.